src/app/PCM/templates/tpl_vertical_cards/index.js

```diff
@@ -0,0 +1,139 @@
+import { useSelector } from 'react-redux'
+
+const TemplateVerticalCards = ({ mappedPcm, depth }) => {
+  const sbeCodePrefix = mappedPcm?.PRI_LOC1 || ''
+  const keys = useSelector(selectKeys)
+  const sbeCode = (filter(key => includes(sbeCodePrefix)(key) && !includes('@')(key))(keys) || [
+    '',
+  ])[0]
+  const actions = useGetActionsFromCode(sbeCode) || []
+  const actionCodes = map(act => act?.attributeCode || '')(actions)
+  const mappedSbe = useGetMappedBaseEntity(sbeCode)
+  const mappedValues = getFields(getColumnDefs(mappedSbe))
+
+  const rows = useSelector(selectCode(sbeCode, 'rows')) || []
+  const { buttonBackgroundColor } = useProductColors()
+
+  return (
+    <>
+      {!!rows.length && (
+        <Box padding={'10px'}>
+          <Box paddingBottom={3}>
+            <Title sbeCode={sbeCode} />
+          </Box>
+
+          <Grid gap={4}>
+            {rows.map(item => (
+              <Card
+                actions={actionCodes}
+                key={`CARD-${item || ''}`}
+                mappedValues={mappedValues}
+                baseEntityCode={item}
+                primaryColor={buttonBackgroundColor}
+                sbeCode={sbeCode}
+              />
+            ))}
+          </Grid>
+        </Box>
+      )}
+    </>
+  )
+}
+
+//Image is hardcoded for the demo, need to remove it after the demo.
+
+const Card = ({ mappedValues, baseEntityCode, actions, sbeCode, primaryColor }) => {
+  return (
+    <Box
+      p={3}
+      bg={'white'}
+      border={`1px solid`}
+      borderRadius="3xl"
+      borderColor={primaryColor}
+      cursor={'pointer'}
+      transition={'all 0.25s ease-in-out'}
+      role="group"
+      _hover={{
+        background: 'product.gradient100',
+      }}
+    >
+      <VStack spacing={2}>
+        {mappedValues.map((value, index) => {
+          const fontSize = index === 0 ? 'xl' : 'md'
+
+          return (
+            <HStack
+              justify={'space-between'}
+              fontSize={fontSize}
+              key={`CARD-ATTRIBUTE-${baseEntityCode}-${value}`}
+              alignSelf="start"
+              color="product.primary"
+              fontWeight="400"
+              w={'full'}
+              position={'relative'}
+              _groupHover={{
+                color: 'white',
+              }}
+            >
+              <Attribute
+                code={baseEntityCode}
+                attribute={value}
+                config={
+                  equals(value)('PRI_IMAGE_URL')
+                    ? {
+                        carddisplay: 'true',
+                        w: 'min(100%, 20rem)',
+                        h: 'auto',
+                        borderRadius: 0,
+                        borderTopLeftRadius: 'xl',
+                        borderTopRightRadius: 'xl',
+                        overflow: 'hidden',
+                      }
+                    : { carddisplay: 'true' }
+                }
+              />
+
+              {index === 0 && (
```

Comment:
This is not the best way to make the check using index as the item in the index can vary at any point, we could add some additional check to this.

Reply:
Once the data from BE gets to FE, I will check this again. For now, I had just copied this from TPL_HORIZONTAL_CARDS.